### PR TITLE
feat: lazy load individual services

### PR DESCRIPTION
**Pull request by ivanahuckova**
Fixes https://github.com/grafana/explore-logs/issues/248

This PR introduces:
- running of individual queries for each service (faster results on screen)
- lazy loading for services
- visually showing which services have been previously selected
- removing LIMIT_SERVICES as we are now lazy loading

https://github.com/grafana/explore-logs/assets/30407135/dce954a1-4e81-40e9-85cd-6a569d0e5ced

### CODE REVIEW

**Review by gtk-grafana — APPROVED** [review]
Code looks good, and it improves greatly on what we have, I would argue we can merge in as is, and address my nits as lower priority issues later.

**Comment by ivanahuckova** [comment]
@gtk-grafana related to

> Edge case: with broken datasource that was previously working, I'd expect to see the configuration UI, but not any previously selected services.

Can you provide steps to reproduce?

**Comment by gtk-grafana** [comment]
@ivanahuckova Sorry about that! 
I was able to repro by simply stopping the docker container, but it can also be reproduced by reverting back to a 2.x version of loki that doesn't support the new endpoints

File: src/Components/Index/ServiceSelectionComponent.tsx [hunk 1]
```diff
@@ -163,118 +154,92 @@ export class ServiceSelectionComponent extends SceneObjectBase<ServiceSelectionC
...
-                    reactNode: <LoadingPlaceholder text="Fetching services..." />,
-                  }),
-                }),
-              ],
-            }),
-            repeatByLabel: SERVICE_NAME,
-            getLayoutChild: this.getLayoutChild.bind(this),
+          new SceneCSSGridLayout({
+            children,
+            isLazy: true,
```
Comment (L2D2Grafana) [inline src/Components/Index/ServiceSelectionComponent.tsx:177]: 🪄 magic!

**Review by L2D2Grafana — APPROVED** [review]
This is great I do think it is best to call `/volume` and then iterate over each label to call the query runner.

File: src/Components/Index/ServiceSelectionComponent.tsx (hunk 1 above)
Comment (ivanahuckova) [reply src/Components/Index/ServiceSelectionComponent.tsx:171]: Yeah, this specifically is a huge benefit of scenes. 🪄

**Pull request merged**